.github/workflows/deploy_to_staging.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will update the code for the staging environment
+
+on:
+  pull_request:
+  push:
+    branches: ["main"]
+
+name: Deploy to staging
+
+jobs:
+  build_and_deploy:
+    name: Deploy
+    environment:
+      name: Staging
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+
+      - name: Set up SAM
+        uses: aws-actions/setup-sam@v1
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Build
+        run: sam build --use-container -t templates/staging.yaml
+
+      - name: Deploy
+        run: sam deploy --stack-name QuizBackendStaging --s3-bucket quiz-staging-backend --no-confirm-changeset --no-fail-on-empty-changeset --region us-east-1 --capabilities CAPABILITY_IAM --parameter-overrides MongoAuthCredentials=${{ secrets.MONGO_AUTH_CREDENTIALS }}
```

Comment:
can you add some comments?

the old deployment steps were super detailed in the README.md
maybe 1-2 lines from the old steps can be added here?